code/object/objectdock.cpp

```diff
@@ -105,8 +105,16 @@ bool dock_check_find_direct_docked_object(object *objp, object *other_objp)
 
 bool dock_check_find_docked_object(object *objp, object *other_objp)
 {
-	Assert(objp != NULL);
-	Assert(other_objp != NULL);
+	Assert(objp != nullptr);
+	Assert(objp->signature > 0);
+	Assert(other_objp != nullptr);
+	Assert(other_objp->signature > 0);
+
+
+	if (!(objp != nullptr && objp->signature > 0))
```

Comment:
I'm pretty sure this check is slightly wrong.  Valid signatures start *at* 0 because they store the index for whatever array that object is a part of, ships, weapons, etc. So this should be >= or > -1.

Net_signatures are "invalid" at 0, but that's another field and only matters during multi.

Welcome to the weirdness of FSO.

Reply:
Signatures start at 1. 0 is invalid:
```c++
int Object_next_signature = 1;	//0 is bogus, start at 1
```

The `instance` starts at 0 and is an index into the relevant subtype array.

Reply:
This is adapted from the scripting API that has the object-is-valid function defined as 
``return (objp != NULL && objp->signature == sig && sig > 0);``
So if that is wrong, the scripting API needs a change too.

Reply:
Wth